src/frontmatter/frontmatter.ts

```diff
@@ -1,15 +1,23 @@
-import { Optional } from '@stoplight/types';
-import * as yaml from 'js-yaml';
+import { DiagnosticSeverity, IDiagnostic, Optional } from '@stoplight/types';
+import { parseWithPointers as parseYaml, safeStringify } from '@stoplight/yaml';
 import { get, pullAt, set, toPath, unset } from 'lodash';
 import * as Unist from 'unist';
 
 import { parseWithPointers } from '../parseWithPointers';
 import { stringify } from '../stringify';
 import { IFrontmatter, PropertyPath } from './types';
 
-const safeParse = (value: string) => {
+const isError = ({ severity }: IDiagnostic) => severity === DiagnosticSeverity.Error;
+
+const safeParse = <T>(value: string): T | {} => {
   try {
-    return yaml.safeLoad(String(value));
+    const { data, diagnostics } = parseYaml<T>(String(value));
+
+    if (data === void 0 || (diagnostics.length > 0 && diagnostics.some(isError))) {
+      return {};
```

Comment:
To be frank, we should return null here

Reply:
Hmm - how often are there diagnostic errors when parsing? And if there are any, does it actually mean that there's no point in returning `data`?

Reply:
It depends on the situation. In certain cases the data might be fairly relevant, while in others it's crap - it's up to the data you pass in.
We might try including `data` and see how it feels.
Thoughts?

Reply:
My intention was to retain the former behavior, yet we can certainly tweak it and return the data we could process

Reply:
ah ok no that's fine